AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Core/SQLiteStorageEngineAdapterTests.swift

```diff
@@ -375,4 +375,120 @@ class SQLiteStorageEngineAdapterTests: BaseDataStoreTests {
             XCTFail(String(describing: error))
         }
     }
+
+    func testClearIfNewVersionShouldStoreInitialVersion() {
+
+        guard let userDefaults = UserDefaults.init(suiteName: "testVersionIsEmpty") else {
+            XCTFail("Could not create a UserDafult with this suite name")
+            return
+        }
+        userDefaults.removeObject(forKey: SQLiteStorageEngineAdapter.dbVersionKey)
+
+        let newVersion = "newVersion"
+
+        let mockFileManager = MockFileManager()
+
+        do {
+            try SQLiteStorageEngineAdapter.clearIfNewVersion(version: newVersion,
+                                                             dbFilePath: URL(string: "dbFilePath")!,
+                                                             userDefaults: userDefaults,
+                                                             fileManager: mockFileManager)
+        } catch {
+            XCTFail("Test has failed due to \(error)")
+        }
+
+        _ = UserDefaults.removeObject(userDefaults)
+    }
+
+    func testVersionInUserDefaultsRemainsSameAndFileExists() {
+        let fileExistsInvoked = expectation(description: "Version remains the same")
+
+        guard let userDefaults = UserDefaults.init(suiteName: "testVersionInUserDefaultsRemainsSame") else {
+            XCTFail("Could not create a UserDafult with this suite name")
+            return
+        }
+
+        userDefaults.set("previousVersion", forKey: SQLiteStorageEngineAdapter.dbVersionKey)
+
+        let newVersion = "previousVersion"
+
+        let mockFileManager = MockFileManager()
+        mockFileManager.fileExists = true
+
+        do {
+            try SQLiteStorageEngineAdapter.clearIfNewVersion(version: newVersion,
+                                                             dbFilePath: URL(string: "dbFilePath")!,
+                                                             userDefaults: userDefaults,
+                                                             fileManager: mockFileManager)
+        } catch {
+            XCTFail("Test has failed due to \(error)")
+        }
+
+        fileExistsInvoked.fulfill()
+
+        wait(for: [fileExistsInvoked], timeout: 10)
```

Comment:
what is this fulfill for?

Reply:
Hmm, I just thought that for every test should have a fulfill but found out that it is not necessary here. Removed